plutus-core/plutus-core/test/Generators/QuickCheck/Utils.hs

```diff
@@ -2,32 +2,55 @@
 
 module Generators.QuickCheck.Utils where
 
+import PlutusPrelude
+
+import PlutusCore.Data
+import PlutusCore.Generators.QuickCheck.Builtin
 import PlutusCore.Generators.QuickCheck.Utils
 
-import Data.Coerce
 import Test.Tasty
 import Test.Tasty.QuickCheck
 
--- | Test that both 'multiSplit1' and 'multiSplit' produce a list such that 'concat'ing it gives
+-- | Test that both 'multiSplit1' and 'multiSplit0' produce a list such that 'concat'ing it gives
 -- back the input.
 test_multiSplitSound :: TestTree
 test_multiSplitSound =
     testGroup "soundness" $ do
         (name, split) <-
             [ ("multiSplit1", coerce $ multiSplit1 @Int)
-            , ("multiSplit", multiSplit 0.1)
+            , ("multiSplit0", multiSplit0 0.1)
             ]
         pure . testProperty name $ \(xs :: [Int]) ->
             withMaxSuccess 10000 . forAll (split xs) $ \aSplit ->
                 xs === concat aSplit
 
--- | Show the distribution of lists generated by 'multiSplit' for a list of the given length.
+-- | Show the distribution of lists generated by a split function for a list of the given length.
 test_listDistribution :: Int -> ([()] -> Gen [[()]]) -> Int -> TestTree
 test_listDistribution numRuns split n =
     testProperty ("for a list of length " ++ show n) $
         withMaxSuccess numRuns . forAll (split $ replicate n ()) $ \aSplit ->
             label (show $ map length aSplit) True
 
+-- | Count the number of 'I' and 'B' nodes in a 'Data' object.
+countIandBs :: Data -> Int
+countIandBs = go 0 where
+    go :: Int -> Data -> Int
+    go acc (Constr _ ds) = foldl' go acc ds
+    go acc (Map ps)      = foldl' (\acc' (d1, d2) -> go (go acc' d1) d2) acc ps
+    go acc (List ds)     = foldl' go acc ds
+    go acc (I _)         = acc + 1
+    go acc (B _)         = acc + 1
+
+-- | Test the number of 'I' and 'B' nodes in a 'Data' generated from a @spine :: [()]@ equals the
+-- length of the spine. Ensures that the 'Data' generator is not exponential in 'B' and 'I' nodes
```

Comment:
I'm not sure why you only care about `B` and `I` not being exponential. A `Data` could be exponentially large without any `B` or `I` at all, right?

Reply:
Yes, but I don't know how to relate a spine to a `Data` object generated from it to ensure there's no exponentiality whatsoever going on. Plus I do insert empty but very much present `Data` nodes when the spine runs out. It's just really non-trivial to control that in the generator, so that you can have a property test for it.

So I thought I'd at least create a test for what I know how to test. Still better than no such tests at all!